### PR TITLE
[RF] Add initial RooFit-AD benchmarks.

### PR DESCRIPTION
**Pull request by grimmmyshini**
Add a simple minimization benchmark for RooFuncWrapper with AD.

### CODE REVIEW

File: root/roofit/roofit-ad/RooFuncWrapperBenchmarks.cxx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+   ->ArgsProduct({{1}, benchmark::CreateRange(10, 10000, 10)})
+   ->Unit(unit)
+   ->Iterations(100);
+
+// Run the minimization with the code-squashed version with AD.
+BENCHMARK(BM_RooFuncWrapper_Minimization)
+   ->Name("NllCodeSquash_AD")
+   ->ArgsProduct({{2}, benchmark::CreateRange(10, 10000, 10)})
+   ->Unit(unit)
+   ->Iterations(100);
```
Comment (vgvassilev) [inline root/roofit/roofit-ad/RooFuncWrapperBenchmarks.cxx:120]: I expect all of the subsequent benchmarks to be comparing against these 4 elements. Can we create some benchmark fixture for these types of benchmarks?

Comment (grimmmyshini) [reply root/roofit/roofit-ad/RooFuncWrapperBenchmarks.cxx:120]: I dont think a benchmark fixture would make sense here but I can move them into a function?

Comment (grimmmyshini) [reply root/roofit/roofit-ad/RooFuncWrapperBenchmarks.cxx]: @vgvassilev I have added a utility function to run all the benchmarks in the new commit.

**Review by guitargeek — APPROVED** [review]
Thank you very much!

It's good to merge the benchmark you have created for CHEP

**Pull request merged**